scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1289,13 +1315,21 @@ private void handleTimeout(DTLSFlight flight) {
 			LOGGER.log(Level.FINE, "Re-transmitting flight for [{0}], [{1}] retransmissions left",
 					new Object[]{flight.getPeerAddress(), max - flight.getTries() - 1});
 
-			flight.incrementTries();
-			sendFlight(flight);
+			try {
+				flight.incrementTries();
+				flight.setNewSequenceNumbers();
```

Comment:
I did not change anything here (at least not intentionally ;-)). We have always used new (record) sequence numbers when re-transmitting a flight. From my point of view this also is in compliance with the DTLS spec.

Section 4.2.2 reads:
> When a (handshake) message is retransmitted, the same message_seq value is used. Note, however, that from the perspective of the DTLS record layer, the retransmission is a new record. This record will have a new DTLSPlaintext.sequence_number value.

This makes sense exactly because duplicate detection is done based on the record sequence number at the record layer. Imagine the following: 

1. Client sends ClientHello using record seq_no 1.
2. Server processes message, marks record seq_no 1 as "processed" and sends its flight.
3. Flight is lost during transmission.
4. Client re-transmits its ClientHello using the same record seq_no 1
5. Server's record layer discards record as duplicate and the handshake stalls...